kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -186,16 +218,25 @@ public void flush(long offset) {
     // this is possible then we will be flushing data to Cassandra
     // that is not yet in the changelog
     final UUID txnid = UUID.randomUUID();
-    final var result = flush(offset, txnid);
-    if (!result.wasApplied()) {
+    final var writeResult = flush(offset, txnid);
+    if (!writeResult.wasApplied()) {
       final OffsetRow stored = client.getOffset(tableName, partition);
       // we were fenced - the only conditional statement is the
       // offset update, so it's the only failure point
       throw new ProcessorStateException(
```

Comment:
I know this isn't new to this PR, but why do we throw a ProcessorStateException here? For things that are fatal to the thread, we should throw an IllegalStateException or something like that. If it just means the thread was fenced and it's possible to recover by rejoining the consumer group (as is the case here right?) then imo we should throw a TaskMigratedException.

Tbh I'm not even clear on the semantics of the ProcessorStateException in Streams, because they're always changing. Might be worth doing a pass over our error handling in general at some point (perhaps we also want our own exception handler)

Reply:
Honestly I had no idea what the different exception types that were being thrown mean and it wasn't totally clear to me so I just threw this one, happy to have a follow-up PR where we clean up all the exceptions.